compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUMmt4dVectorLowering.cpp

```diff
@@ -158,6 +158,8 @@ void LLVMCPUMmt4dVectorLoweringPass::runOnOperation() {
     RewritePatternSet patterns(&getContext());
     mlir::vector::populateVectorTransferDropUnitDimsPatterns(patterns);
     mlir::vector::populateFlattenVectorTransferPatterns(patterns);
+    mlir::vector::populateScalarVectorTransferLoweringPatterns(
```

Comment:
Why is this pass needed here?

Reply:
I think this was an attempt to try some fixes...

Reply:
I tried to fix it because the original error is about scalar types. I remember that you added some features upstream and I give it a shot. It does fix some unit matmul issues but not all of them.